fetch.js

```diff
@@ -337,7 +337,25 @@
   self.Request = Request;
   self.Response = Response;
 
-  self.fetch = function (url, options) {
+  self.fetch = function (input, options) {
+    var url;
+    if (input instanceof Request) {
+      if (input.bodyUsed === true) {
+        throw new TypeError('Body has already been used');
+      }
+
+      url = input.url;
+      options = options || {};
+      options.credentials = options.credentials || input.credentials;
+      options.headers = options.headers || input.headers;
+      options.method = options.method || input.method;
+      options.mode = options.mode || input.mode;
+      options.body = options.body || input._bodyInit;
+    }
+    else {
+      url = input;
+    }
+
```

Comment:
The [`Request` constructor](https://fetch.spec.whatwg.org/#request-class) is responsible for this initialization.